### PR TITLE
[tool] Remove repo install kiwi with pip for Debian and Ubuntu also update config.sh

### PR DESCRIPTION
**Pull request by AvengerMoJo**
Replace the kiwi package install with pip instead. 
Update the config.sh remove baseCleanMount replace the vagrant setup with baseVagrantSetup

Signed-off-by: Alex Lau (AvengerMoJo) <alau@suse.com>

### CODE REVIEW

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
The ubuntu build failed due to missing kiwi-ng, @AvengerMoJo is the usage changed? If true, do you need to update the FZTH too?

**Comment by AvengerMoJo** [comment]
> The ubuntu build failed due to missing kiwi-ng, @AvengerMoJo is the usage changed? If true, do you need to update the FZTH too?

Nope the usage should not be change, instead it should be using the venv pip install version of kiwi-ng which mean if your test script need to run kiwi-ng it should be activate venv first. But I'm not sure how to change that

**Comment by kshtsk** [comment]
> > The ubuntu build failed due to missing kiwi-ng, @AvengerMoJo is the usage changed? If true, do you need to update the FZTH too?
> 
> Nope the usage should not be change, instead it should be using the venv pip install version of kiwi-ng which mean if your test script need to run kiwi-ng it should be activate venv first. But I'm not sure how to change that

Like add source venv activate script in this step of ubuntu wasser config?
```
    - name: building image
      command: |
        cd aquarium
        ./tools/build-image.sh

```

But I am wondering what is the best, source vevn from build-image.sh script or before it.

**Comment by AvengerMoJo** [comment]
> But I am wondering what is the best, source vevn from build-image.sh script or before it.

I think I will do it before running. Just giving user an option to not using it?

**Comment by kshtsk** [comment]
ah... right

**Comment by kshtsk** [comment]
jenkins run tumbleweed

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
```
11:08:18 >>> [ DEBUG   ]: 09:08:18 | Set TMPDIR: /home/ubuntu/.kiwi_boxes
11:08:18 >>> [ DEBUG   ]: 09:08:18 | Calling Qemu: ['/usr/bin/qemu-system-x86_64', '-m', '8096', '-machine', 'accel=kvm', '-cpu', 'host', '-nographic', '-nodefaults', '-snapshot', '-kernel', '/home/ubuntu/.kiwi_boxes/leap/kernel.x86_64', '-append', '"console=hvc0 root=/dev/vda1 rd.plymouth=0 kiwi=\\"--type oem --profile Ceph-Vagrant-libvirt system build\\" sharing-backend=_9p_"', '-drive', 'file=/home/ubuntu/.kiwi_boxes/leap/Leap-Box.x86_64-1.15.2-System-BuildBox.qcow2,if=virtio,driver=qcow2,cache=off,snapshot=on', '-netdev', 'user,id=user0', '-device', 'virtio-net-pci,netdev=user0', '-device', 'virtio-serial', '-chardev', 'stdio,id=virtiocon0', '-device', 'virtconsole,chardev=virtiocon0', '-fsdev', 'local,security_model=mapped,id=fsdev0,path=/home/ubuntu/aquarium/images/build/aquarium', '-device', 'virtio-9p-pci,id=fs0,fsdev=fsdev0,mount_tag=kiwidescription', '-fsdev', 'local,security_model=mapped,id=fsdev1,path=/home/ubuntu/aquarium/images/build/aquarium/_out', '-device', 'virtio-9p-pci,id=fs1,fsdev=fsdev1,mount_tag=kiwibundle', '-initrd', '/home/ubuntu/.kiwi_boxes/leap/initrd.x86_64', '-smp', '4']
11:08:18 EEE Could not access KVM kernel module: Permission denied
11:08:18 EEE qemu-system-x86_64: failed to initialize kvm: Permission denied
```

**Comment by kshtsk** [comment]
this is new

**Comment by AvengerMoJo** [comment]
> this is new

I don't experience this in my system. Maybe adding kvm and libvirt group to user help?

**Comment by kshtsk** [comment]
> > this is new
> 
> I don't experience this in my system. Maybe adding kvm and libvirt group to user help?

you do not experience it because you have started a libvirt probably before building the image, this not the case for openSUSE, building image does not require it I guess. Also, there is a bug in build script which reports that image is built but it is not.

**Comment by AvengerMoJo** [comment]
.jenkins run ubuntu

**Comment by kshtsk** [comment]
the dot before jenkins is extra

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by AvengerMoJo** [comment]
jenkins run ubuntu

**Comment by AvengerMoJo** [comment]
jenkins run ubuntu

**Comment by AvengerMoJo** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by AvengerMoJo** [comment]
> jenkins run ubuntu

@kshtsk how could we add the following to the build host? 
apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb

**Review by jhesketh — DISMISSED** [review]

**Comment by kshtsk** [comment]
> > jenkins run ubuntu
> 
> @kshtsk how could we add the following to the build host?
> apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb

cypress dependencies https://docs.cypress.io/guides/getting-started/installing-cypress#Ubuntu-Debian

**Comment by kshtsk** [comment]
@AvengerMoJo it is tools/setup-dev.sh, see how it is done for cypress_opensuse_leap

File: .wasser/config-ubuntu.yaml [hunk 1]
```diff
@@ -13,15 +13,17 @@ routines:
       command: |
         cd aquarium
         ./tools/setup-dev.sh
-    - name: building image
-      command: |
-        cd aquarium
-        ./tools/build-image.sh
+        sudo adduser ubuntu kvm
```
Comment (kshtsk) [inline .wasser/config-ubuntu.yaml:16]: So after adding this we do not need strictly run tools/build-image.sh after libvirt started?

**Comment by AvengerMoJo** [comment]
jenkins run ubuntu

File: tools/setup-dev.sh [hunk 2]
```diff
@@ -234,15 +244,18 @@ if ! ${skip_install_deps} ; then
             exit 1
           }
           ;;
+        debian | ubuntu)
+          echo "=> try installing dependencies for cypress"
+          sudo apt-get install -q -y ${cypress_debian_ubuntu[*]} || {
+            echo "Dependency installation for cypress failed"
+            exit 1
+          }
+          ;;
```
Comment (kshtsk) [inline tools/setup-dev.sh:253]: This should be moved out of `opensuse-tumbleweed | opensuse-leap` section.

File: tools/setup-dev.sh [hunk 3]
```diff
@@ -234,15 +244,18 @@ if ! ${skip_install_deps} ; then
...
       esac
       ;;
     debian)
       echo "=> installing nodejs16.x repo to apt source"
       wget -qO - https://deb.nodesource.com/setup_16.x | sudo bash -
```
Comment (kshtsk) [inline tools/setup-dev.sh:258]: by the way, I am not sure but cypress does not support node16 jet, does it?

**Comment by AvengerMoJo** [comment]
jenkins run ubuntu

**Comment by AvengerMoJo** [comment]
jenkins run leap

**Comment by AvengerMoJo** [comment]
jenkins run leap

**Review by kshtsk — APPROVED** [review]
Looks good

**Pull request merged**